docs/users-guide/performance/chunking.md

```diff
@@ -0,0 +1,65 @@
+There is a small overhead associated with chunking, but if the total number of
+chunks is relatively small, the overhead is usually outweighed by the benefits
+in reduced memory usage.
+
+Chunking can be used in two ways in ActivitySim: dynamic and explicit.
+
+## Dynamic Chunking
+
+This is the original chunking system in ActivitySim, where model components are
+chunked into pieces that are selected to be approximately optimal for targeting
+a particular memory usage threshold.  The chunk size is determined by "training"
+the model so that it can estimate the memory usage to simulate each chooser handled
+in each component, and then running in "production" mode where the chunk size is
+set to keep the memory usage below the selected threshold, based on the results from
+the training.
+
+To configure chunking behavior, ActivitySim must first be trained with the model
+setup and machine.  To do so, first run the model with ``chunk_training_mode: training``.
+This tracks the amount of memory used by each table by submodel and writes the results
+to a cache file that is then re-used for production runs.  This training mode is
+*significantly* slower than production mode since it does a lot of memory inspection.
+For a training mode run, set ``num_processors`` to about 80% of the available logical
+processors and ``chunk_size``to about 80% of the available RAM.  This will run the
+model and create the ``chunk_cache.csv`` file in the cache directory for reuse.  After
+creating the chunk cache file, the model can be run with ``chunk_training_mode: production``
+and the desired ``num_processors`` and ``chunk_size``.  The model will read the chunk
+cache file from the cache folder, similar to how it reads cached skims if specified.
+The software trains on the size of problem so the cache file can be re-used and
+only needs to be updated due to significant revisions in input file or changes in
+machine specs.  If run in production mode and no cache file is found then ActivitySim falls
+back to training mode.
+
+For more detail on running with dynamic chunking, see [Chunking](chunk_in_detail).
+
+## Explicit Chunking
+
+This is a simpler system that allows the user to specify the number of choosers
+in each chunk explicitly, either as an integer number of choosers per chunk, or
+as a fraction of the overall number of choosers. Although the total amount of
+memory engaged for processing any particular chunk is ignored and there is no
+effort to find a "optimal" chunk size, this system is easier to use
+and understand than dynamic chunking, and in practice has been found to be more
+robust and reliable. It requires no "training" and is activated by setting the
+`chunk_training_mode` configuration setting to `explicit`.
+
+This method for chunking does rely upon model developers to have identified the
+memory-hungry components and to have set reasonable explicit chunk sizes for them.
```

Comment:
Need some description here about how to determine what model run size is -- run with a 10% or so sample single processed with memory profiling on and chunking off, then calculate what the memory needs are.  (Perhaps we should include a little script in the `utilities`portion of the repo to plot the memory profile itself?)

Reply:
@dhensle That's a great idea to include a script in the repo to plot the memory profile.  Can you create it?

Reply:
Just made a commit to this branch and added the little jupyter notebook I have been using to plot the memory profile.  I put it in other_resources/scripts.